File: java-lombok/src/main/java/it/gabrieletondi/telldontaskkata/domain/Category.java
package it.gabrieletondi.telldontaskkata.domain;

import java.math.BigDecimal;

import static java.math.BigDecimal.valueOf;
import static java.math.RoundingMode.HALF_UP;

public class Category {
    private String name;
    private BigDecimal taxPercentage;

    public void setName(String name) {
        this.name = name;
    }

    public void setTaxPercentage(BigDecimal taxPercentage) {
        this.taxPercentage = taxPercentage;
    }

    public BigDecimal taxRate() {
        return taxPercentage.divide(valueOf(100), 2, HALF_UP);
    }
}

File: typescript/src/domain/Category.ts
import bigDecimal from "js-big-decimal";

export default class Category {
    private _name: string | undefined;
    private _taxPercentage: bigDecimal | undefined;

    constructor(name: string, taxPercentage: bigDecimal) {
        this._name = name;
        this._taxPercentage = taxPercentage;
    }

    get name() {
        return this._name;
    }

    set name(name) {
        this._name = name;
    }


    get taxPercentage() {
        return this._taxPercentage;
    }

    set taxPercentage(taxPercentage) {
        this._taxPercentage = taxPercentage;
    }
}